### DISCUSSION TITLE
LES sfs flux

### DISCUSSION THREAD

**Discussion by bronise777** [AMReX-Combustion/PeleC#857] [General]
Martín, M. Pino, U. Piomelli, and G. V. Candler. "Subgrid-Scale Models for Compressible Large-Eddy Simulations." Theoretical and Computational Fluid Dynamics 13, no. 5 (2000): 361–76. https://link.springer.com/article/10.1007/PL00020896
This is the energy equation
![微信截图_20241012204107](https://github.com/user-attachments/assets/99f04e0a-9a26-4b49-9403-5ea4e4a634e6)
![微信截图_20241012204139](https://github.com/user-attachments/assets/3b7c8eb9-c28e-4645-8bf6-16d73147b8c5)
![微信截图_20241012204152](https://github.com/user-attachments/assets/8064cb11-c691-425e-96d9-6bfbded8edcd)
But in the **_pc_smagorinsky_sfs_term function,_**
flx(iv, UEDEN) =    -sigmadx * Uface[0] - sigmady * Uface[1] - sigmadz * Uface[2];
flx(iv, UEDEN) = flx(iv, UEDEN) - cp * Cs2 / PrT * flux_T;
I have two problems, following the above formula **_sigmadx * Uface[0]_** sign seems to be positive and should have a coefficient of 1/2.

**Comment by marchdf** [comment]
hmmmm I would agree on the missing 1/2. And it seems like there is a sign issue. Is it on the Qj term or the Jj term? I am swamped with some things right now but if you have an argument for one or the other, I would be happy to look into it in a couple of days.

**Reply by bronise777** [reply]
The sign issue seems like on the Jj. But in my test, Jj is very small. So this bug only has a little influence.

**Comment by baperry2** [comment]
I believe the term is correct as is, although there are certainly a lot of different possible sign conventions that could be leading to confusion.

First, the stress term for LES should resemble the viscous stress term, and it does. See: [Combustion/PeleC/blob/203736c2009bc114b6aecdfdf40ef46eeb4859f8/Source/Diffterm.H#L447.](https://github.com/AMReX-Combustion/PeleC/blame/development/Source/LES.H#L206) Notably, the sign of the force term and the sign of the energy flux due to that force should be the same. 

I believe the factor of two is an omission from the Martin et al paper. Equation 37 refers to [Knight et al 1998](https://arc.aiaa.org/doi/pdf/10.2514/6.1998-535), specifically Equation 19, which indicates that the factor of 1/2 is accounted for when introducing the model for $J_j$. (note they have also adopted the opposite sign convention for $\tau_{ij})$
![Screenshot 2024-10-22 at 5 06 15 PM](https://github.com/user-attachments/assets/7e3db9e6-fdb2-41c1-a236-ce6de60bb871)

**Reply by bronise777** [reply]
I'm sorry, there are no sign errors here, But the **sigmadx * Uface[0]** seems to be only the first half of the following formula.
![微信截图_20241024193953](https://github.com/user-attachments/assets/43a94d98-00c8-40d3-bc3d-b589fce204c5)
![微信截图_20241024195332](https://github.com/user-attachments/assets/7228f939-9369-444d-8f91-1ada5ce6b0d6)

**Reply by baperry2** [reply]
The sigma term in that equation comes from molecular viscosity. In PeleC, the molecular and LES/turbulent terms are computed separately. The molecular component is computed in `pc_diffusion_flux` in Diffterm.H
